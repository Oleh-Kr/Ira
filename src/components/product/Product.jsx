import React from "react";
import styles from "./Product.module.scss";
import axios from "axios";


const baseURL = "https://api.spoonacular.com/recipes/<id>/information?apiKey=60c1ca27b27b4afbb847decdc9d34f04";

const baseURL1 = "https://api.spoonacular.com/recipes/<id>/priceBreakdownWidget.json?apiKey=60c1ca27b27b4afbb847decdc9d34f04";

export default function Product() {
  const [post, setPost] = React.useState(null);
  const [post1, setPost1] = React.useState(null);


  React.useEffect(() => {
    const receipt_id = localStorage.getItem("card-id");

    axios.get(baseURL.replace("<id>", receipt_id)).then((response) => {
      var pattern = "</b>",
      re = new RegExp(pattern, "g");
      response.data.summary = response.data.summary.replace(/<b>/g, "").replace(re, "")
      setPost(response.data);
    });

    axios.get(baseURL1.replace("<id>", receipt_id)).then((response) => {
      setPost1(response.data);
    });
  }, []);

    React.useEffect(() => {
      if (post)
      {
        post.summary = post.summary.replace("<b>", "").replace("</b>", "")
        
      }
  }, [post])

  if (!post) return null;
  if (!post1) return null;



  return (
    <>
      <div className={styles.heroSection}>
        <div className={styles.leftCol}>
          <h3 className={styles.subhead}>take a rest</h3>
          <div className={styles.heroCta}>
            <p className={styles.productName}>{post.title}</p>
          </div>
          <div className={styles.heroP}>
            <p className={styles.productTextBig}>{post.summary}</p>
          </div>
        </div>
        <img src={post.image} className={styles.heroImg} alt={post.title} />
        <div>
          <p className={styles.productTextSmall}>{post.summary}</p>
        </div>
      </div>
      <div className={styles.featuresSection}>
        <div className={styles.productContainer}>
          <ul className={styles.featuresList}>
            {post1.ingredients.map((ingredient, i) => (
              <li key={i}>{ingredient.name}</li>
            ))}
          </ul>
          <img className={styles.logoImg} src="images/logo.jpg" alt="Logo" />
        </div>
      </div>
    </>
  );
}