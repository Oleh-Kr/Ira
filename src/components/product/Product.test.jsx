import React from "react";
import { act, create } from "react-test-renderer";
import axios from "axios";
import Product from "./Product";
import { MemoryRouter } from "react-router-dom";

jest.mock("axios");

describe("Product component", () => {
  beforeEach(() => {
    axios.get.mockReset();
  });

  test("renders null when post is null", () => {
    const tree = create(
      <MemoryRouter>
        <Product />
      </MemoryRouter>
    );
    expect(tree).toBeNull();
  });

  test("renders null when post1 is null", () => {
    axios.get.mockResolvedValueOnce({ data: { title: "Test Title" } });

    const tree = create(
      <MemoryRouter>
        <Product />
      </MemoryRouter>
    );
    expect(tree).toBeNull();
  });

  test("renders the product details when post and post1 are available", async () => {
    const mockPost = {
      title: "Test Title",
      summary: "Test Summary",
      image: "test-image.jpg",
    };
    const mockPost1 = {
      ingredients: [
        { name: "Ingredient 1" },
        { name: "Ingredient 2" },
        { name: "Ingredient 3" },
      ],
    };

    axios.get.mockResolvedValueOnce({ data: mockPost });
    axios.get.mockResolvedValueOnce({ data: mockPost1 });

    let tree;
    await act(async () => {
      tree = create(
        <MemoryRouter>
          <Product />
        </MemoryRouter>
      );
      await Promise.resolve();
    });

    const heroSection = tree.root.findByProps({ className: "heroSection" });
    expect(heroSection).toBeDefined();

    const productName = tree.root.findByProps({ className: "productName" });
    expect(productName.children).toEqual([mockPost.title]);

    const productTextBig = tree.root.findByProps({ className: "productTextBig" });
    expect(productTextBig.children).toEqual([mockPost.summary]);

    const heroImg = tree.root.findByProps({ className: "heroImg" });
    expect(heroImg.props.src).toEqual(mockPost.image);

    const productTextSmall = tree.root.findByProps({ className: "productTextSmall" });
    expect(productTextSmall.children).toEqual([mockPost.summary]);

    const featuresList = tree.root.findByProps({ className: "featuresList" });
    expect(featuresList.findAllByType("li")).toHaveLength(mockPost1.ingredients.length);

    const logoImg = tree.root.findByProps({ className: "logoImg" });
    expect(logoImg.props.src).toEqual("images/logo.jpg");
  });
});
