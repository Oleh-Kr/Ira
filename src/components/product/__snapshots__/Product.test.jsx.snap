// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Component renders correctly when API calls are successful 1`] = `null`;
